lightning/src/chain/chaininterface.rs

```diff
@@ -65,6 +65,13 @@ pub trait ChainWatchInterface: Sync + Send {
 }
 
 /// An interface to send a transaction to the Bitcoin network.
+///
+/// We may pass non-final transaction (like local HTLC-timeout tx)
+/// in case of channel force-closure due to an offchain anomaly. To warn
+/// being bounce off of every network mempools, implementation MUST
+/// wait until transaction finalization before p2p announcement. To avoid
+/// unecessary transaction origin privacy leak, rebroadcast is the responsibility
```

Comment:
Hmmmm...While I understand the thinking, I think this is in conflict with safety. Especially wth current fixed-fee lightning transactions, if you don't rebroadcast you may have transactions that cannot get into the mempool at some spike but could be confirmed later which you'll lose.

Reply:
Right, fixed-fee being...fixed you should try to brute-force the mempool at every block, at the price of leaking privacy (and only for local commitment transaction with pending HTLCs, if you don't have pending HTLCs, there is no time-sensitiveness). And even with anchor output, you may have the edge-case of no utxo available for CPFP, in this case you should rebroadcast at every block and try the brute-force...